ingest/workflow/snakemake_rules/nextclade.smk

```diff
@@ -0,0 +1,73 @@
+rule nextclade_denvX:
+    """
+    For each type, classify into the appropriate subtype
+    """
+    input:
+        sequences="results/sequences_denv{x}.fasta",
+        dataset="../nextclade_data/denv{x}",
+    output:
+        "data/nextclade_results/nextclade_denv{x}.tsv",
+    threads: 4
+    params:
+        min_length=1000, # E gene length is approximately 1400
+        min_seed_cover=0.1,
+    shell:
+        """
+        nextclade run \
+          --input-dataset {input.dataset} \
+          -j {threads} \
+          --output-tsv {output} \
+          --min-length {params.min_length} \
+          --min-seed-cover {params.min_seed_cover} \
+          --silent \
+          {input.sequences}
+        """
+
+rule join_nextclade_clades:
+    """
+    Merge all the nextclade results into metadata and split metadata
+    """
+    input:
+        metadata="data/metadata_all.tsv",
+        nextclade_denv1="data/nextclade_results/nextclade_denv1.tsv",
+        nextclade_denv2="data/nextclade_results/nextclade_denv2.tsv",
+        nextclade_denv3="data/nextclade_results/nextclade_denv3.tsv",
+        nextclade_denv4="data/nextclade_results/nextclade_denv4.tsv",
+    output:
+        metadata_all="results/metadata_all.tsv",
+        metadata_denv1="results/metadata_denv1.tsv",
+        metadata_denv2="results/metadata_denv2.tsv",
+        metadata_denv3="results/metadata_denv3.tsv",
+        metadata_denv4="results/metadata_denv4.tsv",
+    shell:
+        """
+        echo "genbank_accession,nextclade_subtype,nextclade_type" \
+        | tr ',' '\t' \
+        > results/nextclade_subtype.tsv
+
+        tsv-select -H -f "seqName,clade" {input.nextclade_denv1} \
+        | awk 'NR>1 {{print $0"\tDENV1"}}' \
+        >> results/nextclade_subtype.tsv
+        tsv-select -H -f "seqName,clade" {input.nextclade_denv2} \
+        | awk 'NR>1 {{print $0"\tDENV2"}}' \
+        >> results/nextclade_subtype.tsv
+        tsv-select -H -f "seqName,clade" {input.nextclade_denv3} \
+        | awk 'NR>1 {{print $0"\tDENV3"}}' \
+        >> results/nextclade_subtype.tsv
+        tsv-select -H -f "seqName,clade" {input.nextclade_denv4} \
+        | awk 'NR>1 {{print $0"\tDENV4"}}' \
+        >> results/nextclade_subtype.tsv
+
+        tsv-join -H \
+            --filter-file results/nextclade_subtype.tsv \
+            --key-fields genbank_accession \
+            --append-fields 'nextclade_subtype,nextclade_type' \
+            --write-all ? \
+            {input.metadata} \
+        > {output.metadata_all}
+
+        tsv-filter -H --str-eq ncbi_serotype:denv1 {output.metadata_all} > {output.metadata_denv1}
+        tsv-filter -H --str-eq ncbi_serotype:denv2 {output.metadata_all} > {output.metadata_denv2}
+        tsv-filter -H --str-eq ncbi_serotype:denv3 {output.metadata_all} > {output.metadata_denv3}
+        tsv-filter -H --str-eq ncbi_serotype:denv4 {output.metadata_all} > {output.metadata_denv4}
+        """
```

Comment:
@j23414 and I spoke about this online because I was confused about what was going on here.

Background that was discussed during our chat: The `metadata_all.tsv` has all dengue metadata, including sequences that are not assigned serotypes. The metadata must then be joined with each individual serotype outputs from Nextclade to get the sequence _subtype_. The metadata + subtype TSV is then split back out into individual serotypes to make them available via data.nextstrain.org. 

This rule can be simplified by splitting into 3 different rules. 
1. Aggregates all serotype Nextclade outputs to create a single nextclade_subtype.tsv, i.e. lines 44 - 59.
2. Joins metadata with nextclade_subtype.tsv, i.e. lines 61 - 67
3. Splits out the all metadata + nextclade file into individual serotype files, i.e. lines 69 - 72

Reply:
Thanks for chatting! Refactored and (hopefully) clarified here: 11a5cbcc94770cc62fa4c1bd4f9be5db4d815d61

Reply:
I was wondering if it was possible to deduplicate some of the code in the three rules and ended up pushing up 
https://github.com/nextstrain/dengue/pull/16/commits/831aa0dc872b0ac9e66f1bf567e4760f803ae74e. Feel free to rework it!

Reply:
Thanks for working refactoring and deduplicating the code! Works as expected and looks more streamlined now :D